internal/client/client.go

```diff
@@ -88,11 +88,9 @@ func (c *Config) Validate() error {
 	return nil
 }
 
-type ShutdownHandler func(ctx context.Context)
```

Comment:
Why did you remove this type?

Reply:
It was the part of the simplifying process, I put it back, It look useful. [a6b3aef](https://github.com/hazelcast/hazelcast-go-client/pull/772/commits/a6b3aef218b08ee2bee4501ee356f35c96a4e97e).